fix enhance envoy readiness probe

ramaraochavali opened the pull request:
Fixed Envoy readiness probe which was reverted in https://github.com/istio/istio/pull/17505

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

istio-testing commented:
@ramaraochavali: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
release-test_istio | 94f856ec6ccf5244a62e68c92d7f5dc23e0e4f09 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17637/release-test_istio/1506) | `/test release-test_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali commented:
@howardjohn Found couple of minor things and fixed them
- When checking for port earlier was using Dial without timeout - introduced timeout now
- For all listeners it was trying to ping on LocalAddress but now changed to it to use whatever listener address is.

Can you PTAL? Also Is there a way to test this if it actually resolved the issue?

ramaraochavali commented:
As I have mentioned above, made couple of changes which may help - was suspecting Dial with out timeouts may be causing some
Issue. I am not 100% certain about whether these will fix. That is I was asking if there is a way to test this without merging?
A

howardjohn commented:
sorry didn't see your comment.. last time on your pr immediately after merging a bunch of tests started failing 20% of time or so, so the fact that they didn't and no retest is a good sign. I ran across this by having an empty cluster and spinning up a couple pods, immediately those used high cpu as well as mixer. So I can test that out again with this pr

ramaraochavali commented:
@howardjohn Great. Thanks. Will wait for you test then.

ramaraochavali commented:
/retest

nmittler commented on pilot/cmd/pilot-agent/status/util/listeners.go:
```diff
@@ -106,3 +107,24 @@ func getLocalIPPrefixes() []string {
+	addrs := make([]string, 0)
+	for _, l := range listeners.ListenerStatuses {
+		if l.Name == networking.VirtualOutboundListenerName || l.Name == networking.VirtualInboundListenerName {
+			addrs = append(addrs, fmt.Sprintf("%s:%d", l.LocalAddress.GetSocketAddress().GetAddress(), l.LocalAddress.GetSocketAddress().GetPortValue()))
```
You should probably be a bit more careful here. Check for `nil` along the way.

istio-testing commented:
@ramaraochavali: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-mixer-no_auth_istio | 4d506a690c0337466deda77c2042a3d2bd1eefb9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17637/e2e-mixer-no_auth_istio/1558) | `/test e2e-mixer-no_auth_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali commented:
/test e2e-mixer-no_auth_istio

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-mixer-no_auth_istio | 4d506a690c0337466deda77c2042a3d2bd1eefb9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17637/e2e-mixer-no_auth_istio/1558) | `/test e2e-mixer-no_auth_istio`
integ-telemetry-k8s-tests_istio | 4d506a690c0337466deda77c2042a3d2bd1eefb9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17637/integ-telemetry-k8s-tests_istio/1550) | `/test integ-telemetry-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali commented:
/test integ-telemetry-k8s-tests_istio

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-mixer-no_auth_istio | 4d506a690c0337466deda77c2042a3d2bd1eefb9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17637/e2e-mixer-no_auth_istio/1558) | `/test e2e-mixer-no_auth_istio`
integ-telemetry-k8s-tests_istio | 4d506a690c0337466deda77c2042a3d2bd1eefb9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17637/integ-telemetry-k8s-tests_istio/1550) | `/test integ-telemetry-k8s-tests_istio`
integ-new-install-k8s-tests_istio | 4d506a690c0337466deda77c2042a3d2bd1eefb9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17637/integ-new-install-k8s-tests_istio/1231) | `/test integ-new-install-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-mixer-no_auth_istio | 4d506a690c0337466deda77c2042a3d2bd1eefb9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17637/e2e-mixer-no_auth_istio/1558) | `/test e2e-mixer-no_auth_istio`
integ-telemetry-k8s-tests_istio | 4d506a690c0337466deda77c2042a3d2bd1eefb9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17637/integ-telemetry-k8s-tests_istio/1550) | `/test integ-telemetry-k8s-tests_istio`
integ-new-install-k8s-tests_istio | 4d506a690c0337466deda77c2042a3d2bd1eefb9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17637/integ-new-install-k8s-tests_istio/1231) | `/test integ-new-install-k8s-tests_istio`
integ-mixer-k8s-tests_istio | 4d506a690c0337466deda77c2042a3d2bd1eefb9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/17637/integ-mixer-k8s-tests_istio/1282) | `/test integ-mixer-k8s-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali commented:
/retest

ramaraochavali commented:
@howardjohn Thanks. I will try to reproduce this locally. for now, I am closing this.
